### PR TITLE
Web2py issues/697-2

### PR DESCRIPTION
**Pull request by ProjectGuerilla**
in regard to: https://github.com/web2py/web2py/issues/697
I submitted a prior pull request here: https://github.com/web2py/pydal/pull/123

This is an improvement to that code based on some suggestions from there.

The addition is that the "fix" is wrapped in a try/excpet block that inserts the fix ONLY when the specific error is encountered. This should minimize other issues within the code but fixes the issue in my testing

### CODE REVIEW

**Comment by mdipierro** [comment]
It fails lots of travis tests. :-(

File: pydal/adapters/base.py [hunk 1]
```diff
@@ -1342,7 +1342,10 @@ def represent(self, obj, fieldtype):
             elif not isinstance(obj, (list, tuple)):
                 obj = [obj]
             if field_is_type('list:string'):
-                obj = map(str,obj)
+                try:
+                    obj = map(str,obj)
+                except UnicodeEncodeError:
+                    obj = map(lambbda x:unicode(x).encode(self.db_codec),obj)
```
Comment (gi0baro) [inline pydal/adapters/base.py]: you have a typo in "lambbda" ;)

Comment (ProjectGuerilla) [inline pydal/adapters/base.py]: Is there a way to resubmit a pull with the error fixed or do I just keep making new ones?

**Comment by niphlod** [comment]
once a PR is open, is a pointer to a branch. if you edit your branch and push changes to it, the PR gets "updated" too. I'd rebase before pushing again, though.

**Comment by ProjectGuerilla** [comment]
OK, thanks! I'll figure that out today. Sorry for all the back-and-forth, this is my first effort at contributing to anything so the process is still a little vague.

**Comment by ProjectGuerilla** [comment]
Fixed typo - tests are passing locally

**Comment by ProjectGuerilla** [comment]
I give up...

File: pydal/adapters/base.py (hunk 1 above)
Comment (mdipierro) [inline pydal/adapters/base.py]: If you fix it in the branch of the pull request it updates automatically

**Comment by niphlod** [comment]
mongo is failing for a totally unrelated thing that got merged already. It has nothing to do with this PR.

**Comment by niphlod** [comment]
more specifically, it seems that pymongo3 released a few day ago is uncompatible with the current DAL.

**Comment by ilvalle** [comment]
@niphlod I'm working on fixing mongo adapter, I'll post a PR later today

 Paolo

2015-04-09 20:43 GMT+02:00 niphlod <notifications@github.com>:

**Comment by niphlod** [comment]
no problem. 
it seems that pymongo 3.0 REAALLY didn't care about backward compatibility, which is kinda a PITA. 
I wonder what current users of mongodb will do :D
Anyway, this PR is safe to be merged.

**Comment by mdipierro** [comment]
Please make a new adapter instead if chacging the old one

**Comment by ProjectGuerilla** [comment]
Just for my own excitement - this was my first open source commit!

**Comment by mdipierro** [comment]
Congratulations! :-)

**Pull request merged**